### PR TITLE
[ON HOLD] Documentation: adding new contributors git cheat-sheet

### PR DESCRIPTION
**Pull request by apostasie**
New open-source contributors - well-intended, competent developers that are not used to public contributions, do regularly struggle with some of the git athletics:
https://github.com/containerd/nerdctl/pull/3366#issuecomment-2311082228

Pretty simple stuff that seems obvious for OGs, like squashing, rebasing, managing a commit pile, etc.

This small doc is meant to list the basic stuff, that we can refer people to instead of explaining again and again how this works.

Point being: lowering the bar for contributions is good for the project.

Let me know if this is useful or if you feel people should just RTFM instead :-).

### CODE REVIEW

**Comment by apostasie** [comment]
Not sure why the CI is not happy.
Will try to summon the dark-god of project checks, Sir Phil @estesp.

File: docs/dev/git-cheat.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
+# Git cheat sheet for (busy) contributors
```
Comment (AkihiroSuda) [inline docs/dev/git-cheat.md:1]: Thanks for writing this.

I wonder this should be a standalone website, as the content isn't specific to nerdctl

File: docs/dev/git-cheat.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+## Requirement: DCO and signing commits
```
Comment (AkihiroSuda) [inline docs/dev/git-cheat.md:3]: "DCO" needs explanation

Comment (AkihiroSuda) [reply docs/dev/git-cheat.md:3]: It should be also clarified that there are two kinds of "signing": DCO and GPG(/SSH)

File: docs/dev/git-cheat.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+Always DCO and sign your commits:
+
+```bash
+git -S -s -m "My commit"
```
Comment (AkihiroSuda) [inline docs/dev/git-cheat.md:8]:
```suggestion
git commit -S -s -m "My commit"
```

Also `-S` and `-s` needs explanation

File: docs/dev/git-cheat.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+```
+
+Note: you do NOT need a gpg key to sign commits. You can just use your ssh key.
```
Comment (AkihiroSuda) [inline docs/dev/git-cheat.md:11]: I guess GPG signing is still preferred over SSH.

Comment (AkihiroSuda) [reply docs/dev/git-cheat.md:11]: Does GitHub's Web UI support verifying SSH signs?

File: docs/dev/git-cheat.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+Edit your `~/.gitconfig`:
+
+```
+[user]
+	name = me
+	email = myemail@example.com
+	signingkey = /myhome/.ssh/id_ed25519_sk.pub
```
Comment (AkihiroSuda) [inline docs/dev/git-cheat.md:19]: "_sk" seems unusual

File: docs/dev/git-cheat.md (hunk 1 above)
Comment (AkihiroSuda) [reply docs/dev/git-cheat.md:1]: Or maybe this should be merged into https://github.com/containerd/project/blob/main/CONTRIBUTING.md

**Comment by AkihiroSuda** [comment]
Thanks, but probably this should be merged to https://github.com/containerd/project/blob/main/CONTRIBUTING.md

File: docs/dev/git-cheat.md (hunk 4 above)
Comment (apostasie) [reply docs/dev/git-cheat.md:11]: > Does GitHub's Web UI support verifying SSH signs?

I think so.
I am signing with my ssh key.

> I guess GPG signing is still preferred over SSH.

No idea.

File: docs/dev/git-cheat.md (hunk 5 above)
Comment (apostasie) [reply docs/dev/git-cheat.md:19]: Right.
Using hardware securities key...
Probably same thing apply to edX.
Will make it look more familiar.

File: docs/dev/git-cheat.md (hunk 3 above)
Comment (apostasie) [reply docs/dev/git-cheat.md:8]: Yeah... will do

**Comment by apostasie** [comment]
> Thanks, but probably this should be merged to https://github.com/containerd/project/blob/main/CONTRIBUTING.md

Sure. That works too. Will move it there.

I want this PR to stay open as-is though (because of the project check fail thay may be of interest).

File: docs/dev/git-cheat.md (hunk 1 above)
Comment (fahedouch) [reply docs/dev/git-cheat.md:1]: we may add it to [fay](https://github.com/containerd/nerdctl/blob/main/docs/faq.md) too

**Comment by apostasie** [comment]
Closing.

After some thoughts, this is definitely generic content that could be reused across many projects.

Pushing it there instead:
https://github.com/farcloser/open-source-help-desk/pull/1

Once merged and properly configured, that should become visible under:
https://dev.farcloser.world/open-source-help-desk

Where hopefully we can send people to when they have basic questions about that stuff.

Thank you folks!
Will ping you for review over there if you have time for that.

**Pull request closed without merging**